### PR TITLE
config: support redirect actions

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
This PR adds support for redirect actions.

**Checklist**:
- [ ] add related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

**Comment by calebdoxsey** [comment]
Still need to update the documentation.

File: docs/reference/settings.yaml [hunk 1]
```diff
@@ -1345,13 +1345,34 @@ settings:
...
+      - name: "Redirect"
+        keys: ["redirect"]
+        attributes: |
+          - `yaml`/`json` setting: 'redirect'
+          - Type: object
+          - Optional
+          - Example: `{ "host_redirect": "example.com" }`
+        doc: |
+          `Redirect` is used to redirect incoming requests to a new URL. The `redirect` field is an object with several optional,
+          options:
```
Comment (travisgroth) [inline docs/reference/settings.yaml:1357]:
```suggestion
          `Redirect` is used to redirect incoming requests to a new URL. The `redirect` field is an object with several possible
          options:
```

File: config/policy.go [hunk 2]
```diff
@@ -264,9 +305,15 @@ func (p *Policy) Validate() error {
...
-		return fmt.Errorf("config: policy bad destination url %w", err)
+	switch {
+	case p.To != "":
+		p.Destination, err = urlutil.ParseAndValidateURL(p.To)
+		if err != nil {
+			return fmt.Errorf("config: policy bad destination url %w", err)
+		}
+	case p.Redirect != nil:
+	default:
+		return fmt.Errorf("config: policy must have either a `to` or `redirect`")
```
Comment (travisgroth) [inline config/policy.go:316]: 👍

**Review by travisgroth — APPROVED** [review]
LGTM

**Pull request merged**